File: MainProject/Code.Commons/Constants/HexMeshConstant.cs
using Godot;

namespace Commons.Constants;

/// Copyright (C) 2025 Zhu Xiaohe(aka ZeromaXHe)
/// Author: Zhu XH
/// Date: 2025-04-22 09:34:26
public static class HexMeshConstant
{
    public static readonly Color Weights1 = Colors.Red;
    public static readonly Color Weights2 = Colors.Green;
    public static readonly Color Weights3 = Colors.Blue;

    public static T[] TriArr<T>(T c) => [c, c, c];
    public static T[] QuadArr<T>(T c) => [c, c, c, c];
    public static T[] QuadArr<T>(T c1, T c2) => [c1, c1, c2, c2];
}

File: TerraObserver/TO.FSharp/Domains/Constants/Meshes/HexMeshConstant.fs
namespace TO.FSharp.Domains.Constants.Meshes

open Godot

/// Copyright (C) 2025 Zhu Xiaohe(aka ZeromaXHe)
/// Author: Zhu XH (ZeromaXHe)
/// Date: 2025-06-10 22:12:10
module HexMeshConstant =
    let weights1 = Colors.Red
    let weights2 = Colors.Green
    let weights3 = Colors.Blue
    let triArr<'T> (c: 'T) = [| c; c; c |]
    let quadArr<'T> (c: 'T) = [| c; c; c; c |]
    let quad2Arr<'T> (c1: 'T) (c2: 'T) = [| c1; c1; c2; c2 |]
